CONTRIBUTING.md

```diff
@@ -10,6 +10,28 @@ https://qiskit.org/documentation/contributing_to_qiskit.html
 In addition to the general guidelines there are specific details for
 contributing to terra, these are documented below.
 
+### Contents
+* [Choose an issue to work on](#Choose-an-issue-to-work-on)
+* [Pull request checklist](#pull-request-checklist)
+* [Changelog generation](#changelog-generation)
+* [Release Notes](#release-notes)
+* [Installing Qiskit Terra from source](#installing-qiskit-terra-from-source)
+* [Test](#test)
+  * [Snapshot testing for visualizations](#snapshot-testing-for-visualizations)
+* [Style and Lint](#style-and-lint)
+* [Development Cycle](#development-cycle)
+  * [Branches](#branches)
+  * [Release Cycle](#release-cycle)
+* [Adding deprecation warnings](#adding-deprecation-warnings)
+* [Dealing with git blame ignore list](#dealing-with-the-git-blame-ignore-list)
+
+### Choose an issue to work on
+Qiskit terra uses the following labels to help non-maintainers find issues best suited to their interests and experience level:
+
+* [good first issue](https://github.com/Qiskit/qiskit-terra/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) - these issues are typically the simplest available to work on, perfect for newcomers
+* [help wanted](https://github.com/Qiskit/qiskit-terra/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22) - these issues are potentially low priority work that core maintainers may not have capacity to work on. They are generally more complex than good first issues, a great option for experienced contributors looking for something a bit more challenging
+* [short project](https://github.com/Qiskit/qiskit-terra/issues?q=is%3Aopen+is%3Aissue+label%3A%22short+project%22) - these issues are bigger pieces of work that require greater time commitment. Good options for hackathons, internship projects etc.
```

Comment:
These are great, and I really like the idea of standardizing expectations around labels (both from contributor and maintainer points of view). Would it make sense to include detail here for maintainers about what would qualify for a given label (e.g. `good_first_issue` as a known problem and sketch of an approach vs `help_wanted` where there maybe more investigation and discussion required), or would that be better somewhere else?

Reply:
Yeah I think clarifying for maintainers as well would be really valuable, but I'm not certain this would be the best section 🤔 Let me see if I can come up with some concise phrasing that would work for both contributors and maintainers